sheepdog/transactions/upload/transaction.py

```diff
@@ -276,9 +273,9 @@ def add_entity(self, doc):
             None
         """
         try:
-            entity = UploadEntity(self, self._config)
-            self.entities.append(entity)
+            entity = UploadEntityFactory.create(self, doc, self._config)
```

Comment:
Would it work to put the `entity.parse(doc)` in `UploadEntityFactory.create`?

Reply:
the way the entities were set up previously was to have the parsing separated from their creation, so I kept it that way since I'm still entirely sure the reasons behind splitting that out from init